Venmo/src/test/java/com/braintreepayments/api/VenmoClientUnitTest.java

```diff
@@ -1496,4 +1523,82 @@ public void onVenmoResult_withPaymentContext_withFailedVaultCall_forwardsErrorTo
         verify(listener).onVenmoFailure(error);
         verify(braintreeClient).sendAnalyticsEvent(endsWith("pay-with-venmo.vault.failed"));
     }
+
+    @Test
+    public void setListener_whenPendingBrowserSwitchResultExists_deliversResultToListener_andSetsPendingResultNull() throws JSONException {
+        VenmoAccountNonce nonce = mock(VenmoAccountNonce.class);
+        BraintreeClient braintreeClient = new MockBraintreeClientBuilder()
+                .build();
+
+        BrowserSwitchResult browserSwitchResult = mock(BrowserSwitchResult.class);
+        when(browserSwitchResult.getStatus()).thenReturn(BrowserSwitchStatus.SUCCESS);
+
+        String successUrl = "sample-scheme://x-callback-url/vzero/auth/venmo/success?username=A-Username&payment_method_nonce=fake-venmo-account-nonce";
+        when(browserSwitchResult.getRequestMetadata()).thenReturn(new JSONObject()
+                .put("deepLinkUrl", successUrl)
+        );
+
+        Uri uri = Uri.parse(successUrl);
+        when(browserSwitchResult.getDeepLinkUrl()).thenReturn(uri);
+
+        VenmoApi venmoApi = new MockVenmoApiBuilder()
+                .createNonceFromPaymentContextSuccess(nonce)
+                .build();
+
+        VenmoClient sut = new VenmoClient(activity, lifecycle, braintreeClient, venmoApi, sharedPrefsWriter, deviceInspector);
+        sut.pendingBrowserSwitchResult = browserSwitchResult;
+        sut.setListener(listener);
+
+        verify(listener).onVenmoSuccess(any(VenmoAccountNonce.class));
```

Comment:
```suggestion
        verify(listener).onVenmoSuccess(same(nonce));
```
Nitpick but just a stronger assertion

Reply:
👍 : https://github.com/braintree/braintree_android/pull/911/commits/84aa6b3a0599a1776f36fb0ecbf6ee14090af030